py/dml/deprecations.py

```diff
@@ -0,0 +1,44 @@
+# © 2021-2023 Intel Corporation
+# SPDX-License-Identifier: MPL-2.0
+
+
+deprecations = {}
+
+
+def deprecation(cls):
+    assert hasattr(cls, 'last_api_version')
+    assert hasattr(cls, 'short')
+    deprecations[cls.__name__] = cls
+    return cls
+
+
+@deprecation
+class PORT_PROXY_IFACES:
```

Comment:
This mechanism is somewhat odd -- I like that deprecations are accessible as top-level objects that belong to the module, and that the tag is declared once, and that there is no introspection magic. I'm not really fond of the duck-typing, but it was the easiest way to get the job done. and it doesn't seem to cause scalability issues.

Reply:
The duck typing makes me sad. Is there no more idiomatic way of declaring singletons which are descendant of a central abstract class?

Reply:
It makes me sad too. Making an `ABC` would work, but AFAICS, it would just add a bit of boilerplate without a clear technical gain. But it would of course have some gains on the moral side, so probably worth it.

Reply:
I created an ABC,  it was probably worth it for readability.